src/rp2_common/pico_rand/include/pico/rand.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2022 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef _PICO_RAND_H
+#define _PICO_RAND_H
+
+#include "pico.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** \file pico/rand.h
+ *  \defgroup pico_rand pico_rand
+ *
+ * Random Number Generator API
+ *
+ * This module generates random numbers at runtime via a couple of entropy
+ * sources and uses those sources to modify the state of a 128-bit 'Pseudo
+ * Random Number Generator' implemented in software.
+ *
+ * The random numbers to be supplied are read from the PRNG which is used
+ * to help provide a large number space.
+ *
+ * The main source of entropy at run time is a provided by ROSC 'random bit'
+ * samples but system time is also used to add a little extra entropy.
+ *
+ * All entropy sources are hashed before application to the PRNG state machine.
+ *
+ * Note: The *first* time a random number is requested, the 128-bit PRNG state
+ * must be seeded.  The seed generation takes approximately 1 millisecond with
+ * subsequent random numbers taking between 10 and 20 microseconds to generate.
+ */
+
+// PICO_CONFIG: PICO_RAND_MIN_ROSC_SAMPLE_TIME_US, Define a default minimum time between sampling the ROSC random bit, min=5, max=20, group=pico_rand
```

Comment:
Just out of curiosity, is there a reason for these particular min/max values?
Probably makes sense to also include the `default=10` that you use in the `.c` file?

Reply:
I did a lot of testing using https://github.com/Fourmilab/ent_random_sequence_tester on results collected over a serial port (over many hours : )

And, for whatever reason, observed better results with samples spaced 10us apart of more, than faster samples.
The limits I have specified are something of an estimation, based on the fact that 10us was a decent data point.

Do I have the option not to specify min and max I wonder?

Reply:
Yeah, I think the min and max are only so that https://github.com/raspberrypi/pico-project-generator can display suitable options. Should be fine to leave them off if you want.

Reply:
Thanks.  I might stiffen the text a tiny bit to indicate 10us was as a result of testing.